SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialStatics.cpp

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Utils/SpatialStatics.h"
+
+#include "EngineClasses/SpatialNetDriver.h"
+#include "GeneralProjectSettings.h"
+#include "SpatialConstants.h"
+#include "SpatialGDKSettings.h"
+#include "Utils/ActorGroupManager.h"
+#include "Engine/World.h"
+
+bool USpatialStatics::IsSpatialNetworkingEnabled()
+{
+    return GetDefault<UGeneralProjectSettings>()->bSpatialNetworking;
+}
+
+UActorGroupManager* USpatialStatics::GetActorGroupManager(const UObject* WorldContext)
+{
+	if (const UWorld* World = WorldContext->GetWorld())
+	{
+		if (const USpatialNetDriver* SpatialNetDriver = Cast<USpatialNetDriver>(World->GetNetDriver()))
+		{
+			return SpatialNetDriver->ActorGroupManager;
+		}
+	}
+	return nullptr;
+}
+
+bool USpatialStatics::IsSpatialOffloadingEnabled()
+{
+    return IsSpatialNetworkingEnabled() && GetDefault<USpatialGDKSettings>()->bEnableOffloading;
+}
+
+bool USpatialStatics::IsActorGroupOwnerForActor(const AActor* Actor)
+{
+	if (Actor == nullptr)
+	{
+		return false;
+	}
+
+	return IsActorGroupOwnerForClass(Actor, Actor->GetClass());
+}
+
+bool USpatialStatics::IsActorGroupOwnerForClass(const UObject* WorldContextObject, const TSubclassOf<AActor> ActorClass)
+{
+	if (UActorGroupManager* ActorGroupManager = GetActorGroupManager(WorldContextObject))
+	{
+		const FName ClassWorkerType = ActorGroupManager->GetWorkerTypeForClass(ActorClass);
+		const FName CurrentWorkerType = WorldContextObject->GetWorld()->GetGameInstance()->GetSpatialWorkerType();
+		return (ClassWorkerType == CurrentWorkerType);
+	}
+
+	if (const UWorld* World = WorldContextObject->GetWorld())
+	{
+		return (World->GetNetMode() != NM_Client);
+	}
+
+	return false;
+}
+
+bool USpatialStatics::IsActorGroupOwner(const UObject* WorldContextObject, const FName ActorGroup)
+{
+	if (UActorGroupManager* ActorGroupManager = GetActorGroupManager(WorldContextObject))
+	{
+		const FName ActorGroupWorkerType = ActorGroupManager->GetWorkerTypeForActorGroup(ActorGroup);
+		const FName CurrentWorkerType = WorldContextObject->GetWorld()->GetGameInstance()->GetSpatialWorkerType();
```

Comment:
Can you do a null check on GameInstance? I've had issues with it being nullptr in the engine. If someone used this function in a construction script, it will crash in the editor outside of PIE

Reply:
Guess also do a null check on the world as well